src/main/java/org/sonar/plugins/checkstyle/metadata/CheckUtil.java

```diff
@@ -0,0 +1,207 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.sonar.plugins.checkstyle.metadata;
+
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.reflect.ClassPath;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.BeforeExecutionFileFilter;
+import com.puppycrawl.tools.checkstyle.api.Filter;
+import com.puppycrawl.tools.checkstyle.api.RootModule;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+public final class CheckUtil {
+    private CheckUtil() {
+    }
+
+    /**
+     * Gets all checkstyle's modules.
+     * @return the set of checkstyle's module classes.
+     * @throws IOException if the attempt to read class path resources failed.
+     * @see #isCheckstyleModule(Class)
+     */
+    public static Set<Class<?>> getCheckstyleModules() throws IOException {
+        final Set<Class<?>> checkstyleModules = new HashSet<>();
+
+        final ClassLoader loader = Thread.currentThread()
+                .getContextClassLoader();
+        final ClassPath classpath = ClassPath.from(loader);
+        final String packageName = "com.puppycrawl.tools.checkstyle";
```

Comment:
What about sevntu ?

Reply:
I thought of preparing a PoC model first  with checkstyle checks and get mentors' approval and then tweak further.

Reply:
you do have fully qualified name in meta files - you do not need to scan for classes.
you scan only for metafiles or old configs. get from them fully qualified name and load class by it.
please do.

Reply:
Done